OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -407,4 +413,23 @@ public UnsavedWorkState canOpen() {
             return UnsavedWorkState.OPEN;
         }
     }
+
+    private void autoDiscoverLaTeXPath() {
+        try {
+            String command = (ApplicationUtils.isWindows() ? "where " : "which ") + ApplicationUtils.TEX_NAME;
+            Process p = Runtime.getRuntime().exec(command);
+            p.waitFor();
+
+            File path = new File(new Scanner(p.getInputStream()).nextLine());
+
+            if (path.exists() && path.isFile() && path.canExecute()) {
+                ApplicationSettings.Setting.LATEX_PATH.setValue(path.getParent());
+
+                ApplicationSettings.INSTANCE.save();
+            }
+        } catch (IOException | InterruptedException | NoSuchElementException | IllegalStateException ex) {
+            LOGGER.log("Couldn't find LaTeX installed on your computer.");
+            LOGGER.log("Go to Tools -> Options -> LaTeX to specify LaTeX directory.");

```

Comment:
I'm OK with logging the info to the end users, but we should also print the actual exception (ex) to the console (not to application window) so it can be debugged. Catching exception and ignoring it (so called exception swallowing) is a well known anti-pattern that causes many problems.